diff -ur oomidi/oom/liste/editevent.cpp oomidi.new/oom/liste/editevent.cpp
--- oomidi/oom/liste/editevent.cpp	2012-11-19 14:40:45.860868343 -0800
+++ oomidi.new/oom/liste/editevent.cpp	2012-11-19 16:23:23.892065176 -0800
@@ -639,13 +639,13 @@
 				continue;
 		}
 		MidiController* c = port->midiController(num);
-		isList i = sList.begin();
-		for (; i != sList.end(); ++i)
+		isList q = sList.begin();
+		for (; q != sList.end(); ++q)
 		{
-			if (*i == c->name())
+			if (*q == c->name())
 				break;
 		}
-		if (i == sList.end())
+		if (q == sList.end())
 			sList.push_back(c->name());
 	}
 	MidiController* mc = port->midiController(num);
