--- /usr/lib/python2.7/site-packages/zim/templates.py.bak	2011-02-26 21:44:59.154938031 -0800
+++ /usr/lib/python2.7/site-packages/zim/templates.py	2011-02-26 22:42:28.732418741 -0800
@@ -741,10 +741,13 @@
 	def links(self):
 		for type, name, _ in self._page.get_links():
 			if type == 'page':
-				page = self._notebook.get_page(name)
-				yield PageProxy(
-					self._notebook, page,
-					self._format, self._linker, self._options)
+				try:
+					page = self._notebook.get_page(name)
+					yield PageProxy(
+						self._notebook, page,
+						self._format, self._linker, self._options)
+				except:
+					pass
 
 	@property
 	def backlinks(self):
